tests/ALCProxy.Tests/ALCProxyTests.cs

```diff
@@ -0,0 +1,353 @@
+        }
+        [Fact]
+        public void ReturnTypesCorrectlyGiveBackUserDefinedTypes()
+        {
+            TestAssemblyLoadContext alc = new TestAssemblyLoadContext("ReturnTypesCorrectlyGiveBackUserDefinedTypes", isCollectible: true);
+            ITest t = ProxyBuilder<ITest>.CreateInstanceAndUnwrap(alc, Assembly.GetExecutingAssembly().GetName(true), "Test");
+            Assert.Equal(new Test2().ToString(), t.ReturnUserType().ToString());
+
+            alc.Unload();
+        }
+        [Fact]
+        public void CreateObjectWithNonUserParamsInConstructor()
+        {
+            TestAssemblyLoadContext alc = new TestAssemblyLoadContext("CreateObjectWithNonUserParamsInConstructor", isCollectible: true);
+            ITest t = ProxyBuilder<ITest>.CreateInstanceAndUnwrap(alc, Assembly.GetExecutingAssembly().GetName(true), "TestObjectParamConst", new object[] { 55, "testString" });
+            Assert.Equal(55, t.ReturnIntWhilePassingInList(3, new List<string>()));
+            Assert.Equal(58, t.ReturnIntWhilePassingInUserType(3, new Test2()));
+            Assert.Equal("testString", t.GetContextName());
+            Assert.Equal(5, t.SimpleMethodReturnsInt());
+        }
+        [Fact]
+        //TODO currently this breaks due to some type cast errors ("Test2 isn't the same type as Test2"). This may be due to how we load the assemblies
+        //into the ALC, but we need to investigate this further to figure out what's going wrong.
+        public void CreateObjectWithUserParamsInConstructor()
+        {
+            TestAssemblyLoadContext alc = new TestAssemblyLoadContext("CreateObjectWithUserParamsInConstructor", isCollectible: true);
+            //Test when putting user objects into the constructor
+            Test2 test = new Test2(100);
+            ITest t2 = ProxyBuilder<ITest>.CreateInstanceAndUnwrap(alc, Assembly.GetExecutingAssembly().GetName(true), "TestObjectParamConst", new object[] { 60, "testString", test });
+            Assert.Equal(60, t2.ReturnIntWhilePassingInList(3, new List<string>()));
+            Assert.Equal(63, t2.ReturnIntWhilePassingInUserType(3, new Test2()));
+            Assert.Equal("testString", t2.GetContextName());
+            Assert.NotEqual(test, t2.ReturnUserType());
+            Assert.Equal(test.testValue, t2.SimpleMethodReturnsInt());
+        }
+        //As a note, to test these, we need to manually build any of the external classes before running tests, 
+        //or you will error out with a TypeLoadException when the creation of the proxy is attempted
+        [Fact]
+        public void CanLoadOustideAssemblyWithSharedInterface()
+        {
+            string pathOfCurrentString = "ALCProxy.Tests";
+            string assemblyLocation = Assembly.GetExecutingAssembly().Location;
+            string newPath = assemblyLocation.Substring(0, assemblyLocation.IndexOf(pathOfCurrentString)) + "ALCProxy.TestAssembly\\bin\\" + dbgString + "\\netcoreapp3.0\\ALCProxy.TestAssembly.dll";
```

Comment:
Use `Path.Combine` or `Path.DirectorySeparatorChar`, and `Path.GetDirectoryName`